docs/rfcs/0002-ecommerce.md

```diff
@@ -0,0 +1,176 @@
+## Title for RFC
+ecommerce in mitxpro
+
+
+#### Abstract
+
+In MicroMasters we made some decisions about ecommerce. This RFC looks
+at the decisions made in MicroMasters and those made for edX's ecommerce
+stack and makes recommendations for ecommerce in mitxpro.
+
+#### Payment Processors
+
+The only supported payment processor for MicroMasters is CyberSource.
+At the time MIT didn't support other payment processors
+like Stripe for example, and I believe that's still true today.
+
+edX has support for multiple payment processors, for example 
+the [CyberSource payment processor backend](https://github.com/edx/ecommerce/blob/master/ecommerce/extensions/payment/processors/cybersource.py#L369).
+This adds some complexity since there are significant differences
+between how they interact with the web application's frontend and backend.
+
+##### Recommendation
+
+Unless MIT will support other payment processors in the near future, we should
+design to use CyberSource similar to what we do in MicroMasters.
+
+#### Models
+
+MicroMasters only allows a user to purchase one course at a time so there
+is no concept of a cart or basket. In MicroMasters the `Order` object stores
+the information about the order and has a foreign key to `Line` to store
+information about multiple purchased items. However since the user can only purchase one course at
+once, there is only one `Line` per `Order` in MicroMasters.
+
+It's a requirement for mitxpro to support multiple items per order so we'll need a cart or basket
+for the user to keep track of their order. edX uses a `Basket` object for this purpose, which subclasses
+the `AbstractBasket` class from django-oscar. When a user clicks the button to verify a course, they
+are redirected to the `basket/add/?sku=SKU` URL which is handled by django-oscar to add the
+item to the basket.
+
+##### Recommendation
+
+Whether or not we want to use django-oscar, we will need models like these (using django-oscar names here):
+ - `Order` for an instance of a purchase
+ - `Basket` for a user's shopping cart. This should be connected to `Order`.
+ - `Line` for an item in an `Basket`
+ - `Product` to represent a purchasable item. In MicroMasters we skip this and 
+ instead set the course key in the `Line` directly but should have this object
+ to provide future flexibility and to let us decouple the price and the course information.
+ - `Voucher` to describe a coupon and how it can be used. A foreign key to `Basket` can be used
```

Comment:
Would `Voucher` be akin to a promo code for a sale on some ecommerce website? If so, it could apply to many different `Basket`s

Reply:
That's a good point. Maybe we should have a `VoucherBasket` model to map the relationship instead, which would have foreign keys to `Voucher` and `Basket`. We probably need something similar to map `Voucher` to `Product` so we know where it applies.

Reply:
If we can, I'd prefer to call these `Coupon`, lest we get very confused. For the xPro team, a `voucher` is something completely different and, hopefully, not something we need to handle our code at all. 

I plan to check on the use case of coupons that can be used multiple times -- is that what you're asking about @gsidebo ?

Reply:
I am fine calling it `Coupon`. `Voucher` is the model used by django-oscar but there's no restriction on how we name it on our system.